src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -98,15 +106,19 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       response.sendRedirect("/");
       return;
     }
-
     String username = userService.getCurrentUser().getEmail();
 
     String recipient = request.getParameter("recipient");
 
     String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    String regex = "(https?://\\S+\\.(png|jpg|gif))";
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
```

Comment:
This line breaks replies, because they don't go through Blobstore. One quick way to fix this is to wrap it in a try-catch block and catch the exception. Another option is to send the replies through Blobstore.

It's up to you (and Amanda and DJ) whether you want to fix that in this PR or in a later PR.